subprojects/core/src/integTest/groovy/org/gradle/caching/internal/packaging/BuildCacheEntryPackingIntegrationTest.groovy

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal.packaging
+
+import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
+import org.gradle.integtests.fixtures.daemon.DaemonIntegrationSpec
+import spock.lang.Issue
+import spock.lang.Unroll
+
+class BuildCacheEntryPackingIntegrationTest extends DaemonIntegrationSpec implements DirectoryBuildCacheFixture {
+
+    private static final NON_ASCII_NAME = [
+        "ascii-only": "ascii",
+        "space": " ",
+        "zwnj": "\u200c",
+        "chinese": "敏捷的棕色狐狸跳过了懒狗",
+        "cyrillic": "здравствуйте",
+        "hungarian": "Árvíztűrő tükörfúrógép",
+    ].values().join("-")
+
+    private static final DEFAULT_ENCODINGS = [
+        "UTF-8",
+        "ISO-8859-1",
+        "windows-1250",
+    ]
+
+    @Unroll
+    def "can store and load files having non-ascii characters in file name using default file encoding #fileEncoding"() {
+        def fileName = NON_ASCII_NAME + ".txt"
+        def outputFile = file("dir", fileName)
+
+        buildFile << """
+            println "> Default charset: \${java.nio.charset.Charset.defaultCharset()}"
+            println "> Storing file in cache: $fileName"
```

Comment:
❓ does this still make sense or was it added for debugging purposes?

Reply:
If you don't feel strongly about removing this I'd leave this in. It's nice as a human to see this in the output.

Reply:
👍